spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -3149,6 +3189,20 @@ def keyPressEvent(self, event):
             # could be shortcuts
             event.accept()
 
+    def fix_and_strip_indent(self, comment_or_string=False):
+        """Automatically fix indent and strip previous automatic indent."""
+        # Fix indent
+        cursor_before = self.textCursor().position()
+        # A change just occured on the last line (return was pressed)
+        if cursor_before > 0:
+            self.last_change_position = cursor_before - 1
+        self.fix_indent(comment_or_string=comment_or_string)
+        cursor_after = self.textCursor().position()
+        # Remove previous spaces and update last_auto_indent
+        nspaces_removed = self.strip_trailing_spaces()
+        self.last_auto_indent = (cursor_before - nspaces_removed,
+                                 cursor_after - nspaces_removed)

```

Comment:
This should also call `self.document_did_change()` at this point for linting to be updated.

Reply:
done